### PR TITLE
Add javadoc and tests for functional ReceivedMessage class

### PR DESCRIPTION
**Pull request by mziccard**

### CODE REVIEW

File: gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      .attributes(ATTRIBUTES)
+      .publishTime(PUBLISH_TIME)
+      .build();
+  private static final com.google.pubsub.v1.ReceivedMessage RECEIVED_MESSAGE_PB =
+      com.google.pubsub.v1.ReceivedMessage.newBuilder()
+          .setMessage(MESSAGE.toPb())
+          .setAckId(ACK_ID)
+          .build();
+
+  private PubSub serviceMockReturnsOptions = createStrictMock(PubSub.class);
```
Comment (mziccard) [inline gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java]: Codacy Issue found: [Private field 'serviceMockReturnsOptions' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2671233162/issues/source?bid=3123109&fileBranchId=3332594#l65)

File: gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  private PubSubOptions mockOptions = createMock(PubSubOptions.class);
```
Comment (mziccard) [inline gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java]: Codacy Issue found: [Private field 'mockOptions' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2671233162/issues/source?bid=3123109&fileBranchId=3332594#l66)

File: gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    replay(pubsub);
+    initializeMessage();
+    message.ack();
+  }
+
+  @Test
+  public void testAckAsync() throws ExecutionException, InterruptedException {
+    initializeExpectedMessage(1);
+    expect(pubsub.options()).andReturn(mockOptions);
+    EasyMock.expect(pubsub.ackAsync(SUBSCRIPTION, ACK_ID))
```
Comment (mziccard) [inline gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'EasyMock.expect' due to existing static import 'org.easymock.EasyMock.expect'](https://www.codacy.com/app/mziccard/gcloud-java/file/2671233162/issues/source?bid=3123109&fileBranchId=3332594#l146)

File: gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    replay(pubsub);
+    initializeMessage();
+    message.modifyAckDeadline(10, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testModifyAckDeadlineAsync() throws ExecutionException, InterruptedException {
+    initializeExpectedMessage(1);
+    expect(pubsub.options()).andReturn(mockOptions);
+    EasyMock.expect(pubsub.modifyAckDeadlineAsync(SUBSCRIPTION, 10, TimeUnit.SECONDS, ACK_ID))
```
Comment (mziccard) [inline gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'EasyMock.expect' due to existing static import 'org.easymock.EasyMock.expect'](https://www.codacy.com/app/mziccard/gcloud-java/file/2671233162/issues/source?bid=3123109&fileBranchId=3332594#l169)

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/Message.java [hunk 5]
```diff
@@ -244,17 +244,24 @@ public ByteArray payload() {
...
         || obj != null
         && obj.getClass().equals(Message.class)
-        && Objects.equals(toPb(), ((Message) obj).toPb());
+        && baseEquals((Message) obj);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(serialVersionUID, id, payload, attributes, publishTime);
+    return Objects.hash(id, payload, attributes, publishTime);
```
Comment (lesv) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/Message.java]: Why did you remove the serialVersionUID?

**Comment by lesv** [comment]
LGTM

File: gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/Message.java (hunk 5 above)
Comment (mziccard) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/Message.java]: Does it really make sense to have the `serialVersionUID` is `hashCode`? Maybe @aozarov did that for a reason?

Comment (aozarov) [inline gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/Message.java]: No, I did not and +1 for removing it. Was probably done when I was experimenting with one of the IntelliJ hashCode generator based on Java 7 Objects file.

**Comment by mziccard** [comment]
Thanks @lesv for the pass.
